### PR TITLE
Fix warnings in unit conversions (#1047)

### PR DESCRIPTION
**Pull request by darienmorrow**
## Changes

Unit conversions of None to None no longer throws a warning.
Invalid unit conversions now throw a pint error.

## Checklist

- [x] added tests, if applicable
- [x] updated documentation, if applicable
- [x] updated CHANGELOG.md
- [ ] tests pass

### CODE REVIEW

**Comment by darienmorrow** [comment]
I believe this PR has the desired properties as discussed in #1047. 

@ksunden Can you test this against acquisition hardware? 

Example code showing the new behavior is as follows:
```
import WrightTools as wt
import pint

c = wt.units.converter
# fine return converted value
print(c(val=12500, current_unit='wn', destination_unit='eV'))
# fine return input
print(c(val=12500, current_unit=None, destination_unit=None))
# warning return input
print(c(val=12500, current_unit=None, destination_unit='eV'))
print(c(val=12500, current_unit='wn', destination_unit=None))
# error
try: 
    print(c(val=12500, current_unit='wn', destination_unit='W'))
except pint.errors.DimensionalityError:
    print('error thrown 1')
try:
    print(c(val=12500, current_unit='W', destination_unit='wn'))
except pint.errors.DimensionalityError:
    print('error thrown 2')
```

**Comment by darienmorrow** [comment]
ping @ksunden

File: WrightTools/units.py [hunk 1]
```diff
@@ -111,12 +116,15 @@ def converter(val, current_unit, destination_unit):
     number
         Converted value.
     """
-    try:
-        val = ureg.Quantity(val, current_unit).to(destination_unit).magnitude
-    except (pint.errors.DimensionalityError, pint.errors.UndefinedUnitError, AttributeError):
+    if (current_unit == None) and (destination_unit == None):
```
Comment (ksunden) [inline WrightTools/units.py:119]: please use `is None` instead of `== None` (this is more efficient and standard style (see code scanning results)

**Review by ksunden — CHANGES_REQUESTED** [review]
Implementation looks good, works fine with yaqc-cmds.

Please add tests of new behavior (your comment plus some assertions is a good place to start)

**Comment by darienmorrow** [comment]
@ksunden I added tests. If all is good, please merge.

**Comment by ksunden** [comment]
For future reference, @darienmorrow as much as I saw the comment originally that it was ready for review and I resolved some merge conflicts and told myself I would merge once the tests passed.... I ignored this for longer than I intended to in large part due to the WIP in the title (every time I saw it in my notifications list I was like "Oh, this must still need something")

Changing the title would have gotten my attention sooner

**Review by ksunden — APPROVED** [review]

**Pull request merged**